tests/flows/send_specific_receipts.go

```diff
@@ -0,0 +1,248 @@
+		// Get the Teleporter message from receive event and confirm that the receipts are delivered again
+		receiveEvent, err :=
+			utils.GetEventFromLogs(receipt.Logs, subnetAInfo.TeleporterMessenger.ParseReceiveCrossChainMessage)
+		Expect(err).Should(BeNil())
+		log.Info("Receipt included", "count", len(receiveEvent.Message.Receipts), "receipts", receiveEvent.Message.Receipts)
+		Expect(receiptIncluded(messageID1, receiveEvent.Message.Receipts)).Should(BeTrue())
+		Expect(receiptIncluded(messageID2, receiveEvent.Message.Receipts)).Should(BeTrue())
+
+		// Check the reward amount remains the same
+		checkExpectedRewardAmounts(subnetAInfo, receiveEvent1, receiveEvent2, mockTokenAddress, relayerFeePerMessage)
+	}
+}
+
+func clearReceiptQueue(
+	ctx context.Context,
+	network interfaces.Network,
+	fundedKey *ecdsa.PrivateKey,
+	source interfaces.SubnetTestInfo,
+	destination interfaces.SubnetTestInfo,
+) {
+	outstandReceiptCount := getOutstandingReceiptCount(source, destination.BlockchainID)
+	for outstandReceiptCount.Cmp(big.NewInt(0)) != 0 {
+		log.Info("Emptying receipt queue", "remainingReceipts", outstandReceiptCount.String())
+		// Send message from Subnet B to Subnet A to trigger the "regular" method of delivering receipts.
+		// The next message from B->A will contain the same receipts that were manually sent in the above steps,
+		// but they should not be processed again on Subnet A.
+		sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+			DestinationBlockchainID: destination.BlockchainID,
+			DestinationAddress:      common.HexToAddress("0x1111111111111111111111111111111111111111"),
+			RequiredGasLimit:        big.NewInt(1),
+			FeeInfo: teleportermessenger.TeleporterFeeInfo{
+				FeeTokenAddress: common.Address{},
+				Amount:          big.NewInt(0),
+			},
+			AllowedRelayerAddresses: []common.Address{},
+			Message:                 []byte{1, 2, 3, 4},
+		}
+
+		// This message will also have the same receipts as the previous message
+		receipt, _ := utils.SendCrossChainMessageAndWaitForAcceptance(
+			ctx, source, destination, sendCrossChainMessageInput, fundedKey)
+
+		// Relay message
+		network.RelayMessage(ctx, receipt, source, destination, true)
+
+		outstandReceiptCount = getOutstandingReceiptCount(source, destination.BlockchainID)
+	}
+	log.Info("Receipt queue emptied")
+}
+
+func getOutstandingReceiptCount(source interfaces.SubnetTestInfo, destinationBlockchainID ids.ID) *big.Int {
+	size, err := source.TeleporterMessenger.GetReceiptQueueSize(&bind.CallOpts{}, destinationBlockchainID)
+	Expect(err).Should(BeNil())
+	return size
+}
+
+// Checks the given message ID is included in the list of receipts.
+func receiptIncluded(
+	expectedMessageID *big.Int,
+	receipts []teleportermessenger.TeleporterMessageReceipt,
+) bool {
+	for _, receipt := range receipts {
+		if receipt.ReceivedMessageID.Cmp(expectedMessageID) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func checkExpectedRewardAmounts(
```

Comment:
Can we either rename this to something that mentions it's for a round-trip or leave a comment saying that this checks the cumulative rewards for two separate receive events for the same tokenAddress?

Reply:
I figured I might be wrong but thought it would help illustrate my point 😆